subprojects/code-quality/src/test/groovy/org/gradle/api/plugins/quality/CheckstylePluginMultiProjectTest.groovy

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2011 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.plugins.quality
+
+import org.gradle.api.internal.project.ProjectInternal
+import org.gradle.test.fixtures.AbstractProjectBuilderSpec
+import org.gradle.util.TestUtil
+
+class CheckstylePluginMultiProjectTest extends AbstractProjectBuilderSpec {
+
+    def "configures checkstyle extension"() {
```

Comment:
Needs to also provide a test case for a deeply nested project structure. Would you mind adding it?

Reply:
I transformed this into a deeply nested project and the test passes.